packages/mysql/src/mysql-adapter.ts

```diff
@@ -251,11 +251,21 @@ export class MySQLPersistence extends SQLPersistence {
             valuesSetNames.push('_changed_' + fieldName);
         }
 
+        const changedJsonFields = prepared.changedFields.filter(v => {
+            const fieldType = classSchema.getProperty(v).type;
+            const typeMapping = this.platform.getTypeMapping(fieldType);
+            return typeMapping?.sqlType === 'json';
+        });
+
         for (let i = 0; i < changeSets.length; i++) {
             params.push(prepared.primaryKeys[i]);
             let pkValue = placeholderStrategy.getPlaceholder();
             valuesValues.push('ROW(' + pkValue + ',' + prepared.changedFields.map(name => {
-                params.push(prepared.values[name][i]);
+                const val = prepared.values[name][i];
+                if (val === null && changedJsonFields.includes(name)) {
+                    return 'CAST(NULL AS JSON)';
```

Comment:
when a type like `doc: { name: string } | null = null;` is defined we make the column type json and NULLABLE. Unfortunately this expression `CAST(NULL AS JSON)` doesn't change the column value to NULL but to JSON NULL, which is a bit confusing but they are different and we actually what the real database NULL

Reply:
oh, this was a rendering issue. It seems they are the same